Processa uma página por thread

rbelem opened the pull request.

rbelem committed:
feat: Extrai cada página em uma thread separada

Remove também a ProgressBar por não ter suporte a multithread.

rbelem committed:
feat: Adiciona suporte ao pdftk e pdfinfo

* Dockerfile
* Makefile target: split-pdf

rbelem committed:
feat: Adiciona target para concaternar os arquivos csv

rbelem committed:
feat: Adiciona coluna id

rbelem committed:
feat: Processa chunk de páginas

pvfrota commented:
@rbelem Esse erro tá acontecendo com vc tb?

```shell
➜  vacina-manaus-backend git:(process-page-per-thread) make process-data       
Creating vacina-manaus-backend_vacina-manaus-backend_run ... done
Traceback (most recent call last):                                                                                                                                                                                |
  File "src/process_data.py", line 402, in <module>
    dataProcessor.process_all()
  File "src/process_data.py", line 386, in process_all
    df_to_extract().to_csv(
  File "src/process_data.py", line 65, in vaccination_site_count
    return self.df[['vaccination_site', 'id']]\
  File "/usr/local/lib/python3.8/site-packages/pandas/core/frame.py", line 5452, in sort_values
    k = self._get_label_or_level_values(by, axis=axis)
  File "/usr/local/lib/python3.8/site-packages/pandas/core/generic.py", line 1698, in _get_label_or_level_values
    raise ValueError(
ValueError: The column label 'count' is not unique.
ERROR: 1
Makefile:66: recipe for target 'process-data' failed
make: *** [process-data] Error 1
```

Vou dar uma corrigida nessas métricas

pvfrota commented:
https://github.com/CodeForManaus/vacina-manaus-backend/pull/84#issuecomment-779901165

Entendi pq esse erro acontecia, os `.csvs` anteriores a esse PR incluíam a coluna ID, que agora é gerada pelo próprio Pandas.

Sendo assim, não é um bug, mas sim um comportamento esperado :raised_hands:

rbelem committed:
feat: Processa chunks de uma ou múltiplas páginas

rbelem committed:
feat: Substitui split do pdf em chunks por split por páginas

O tempo do split em chunks levava mais de meia hora, enquanto o split
por página leva menos de 1 minuto.

rbelem committed:
feat: Agrupa todos os targets de extract no target extract-data

rbelem committed:
feat: Adiciona workaround para consertar datas inválidas

rbelem committed:
feat: Verifica se linha está em branco e então não adiciona

emoryr reviewed (approved):
LGTM

pvfrota commented:
![](https://media.tenor.com/images/4f0218702ac12d501a826b0e14318ff4/tenor.gif)
